cas-server-documentation/installation/Configuring-SAML2-Authentication.md

```diff
@@ -0,0 +1,165 @@
+---
+layout: default
+title: CAS - SAML2 Authentication
+---
+
+# SAML2 Authentication
+CAS can act as a SAML2 identity provider accepting authentication requests and producing SAML assertions.
+
+## Overview
+
+### SAML Endpoints
+The following CAS endpoints respond to supported SAML2 profiles:
+
+- `/cas/idp/profile/SAML2/Redirect/SSO`
+- `/cas/idp/profile/SAML2/POST/SSO`
+
+### IdP Metadata
+The following CAS endpoints handle the generation of SAML2 metadata:
+ 
+- `/cas/idp/metadata`
+
+This endpoint will display the CAS IdP SAML2 metadata upon receiving a GET request. If metadata is already available and generated,
+it will be displayed. If metadata is absent, one will be generated automatically. 
+CAS configuration below dictates where metadata files/keys will be generated and stored.
+
+<div class="alert alert-info"><strong>Review Metadata</strong><p>Due to the way CAS handles the generation of metadata via external 
+libraries, the generated metadata MUST be reviewed and massaged slightly to match the CAS configuration. All other elements MUST be 
```

Comment:
No. If you are starting from scratch, you access the endpoint to generate your metadata. That's all. If you do lose your metadata for some reason, meaning the file is removed from your metadata location, renamed, etc for whatever reason then the endpoint will regenerate them for you when accessed. If the file remains in the same place, the endpoint will simply show your metadata to you.